### PR TITLE
Fix most tests for windows computers

### PR DESCRIPTION
**Pull request by Miauwkeru**
- Check whether effective_ids is enabled for the system
- Use __type__ consistently
- Open files as UTF-8 by default
- Fix paths in test_target.py
- Change all occurrences to flow.record path to self.target.fs.path
- Fix issues regarding the resolver in trusteddocs

### CODE REVIEW

File: dissect/target/helpers/cache.py [hunk 1]
```diff
@@ -142,9 +142,11 @@ def call(self, *args, **kwargs):
         elif REWRITE_CACHE:
             read_file_cache = False
 
+        effective_ids = True if os.supports_effective_ids else False
```
Comment (Schamper) [inline dissect/target/helpers/cache.py:145]: Can't you just put `effective_ids=os.supports_effective_ids` in all the calls?

File: dissect/target/plugins/os/unix/linux/nixos/_os.py
Comment (Schamper) [inline dissect/target/plugins/os/unix/linux/nixos/_os.py:1]: Accidentally added or intentional addition? Missing a `__init__.py` if the latter case.

File: dissect/target/plugins/os/windows/regf/trusteddocs.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def trusteddocs(self) -> Iterator[TrustedDocumentsRecord]:
                     ts=key.ts,
                     type=value.type,
                     application=application,
-                    document_path=self.target.resolve(value.name),
+                    document_path=self.target.fs.path(self.target.resolve(value.name)),
```
Comment (Schamper) [inline dissect/target/plugins/os/windows/regf/trusteddocs.py:76]: Perhaps `.resolve()` should return a path?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/plugins/os/windows/startupinfo.py
Comment (Poeloe) [inline dissect/target/plugins/os/windows/startupinfo.py:1]: I guess you can remove the `from flow.record.fieldtypes import path` here as well, right?

**Review by Poeloe — CHANGES_REQUESTED** [review]

**Review by Poeloe — APPROVED** [review]
LGTM

**Pull request merged**